action.yaml

```diff
@@ -29,20 +29,23 @@ runs:
       sudo pacman-key --init
       sudo pacman-key --populate archlinux
 
-  # Try to eliminate "Failed to dissect image: Connection timed out" errors from nspawn by compiling
-  # systemd-nspawn from v251 from source.
-  - name: Update systemd-nspawn
+  - name: Update systemd
     shell: bash
     working-directory: ${{ github.action_path }}
     run: |
       echo "deb-src http://archive.ubuntu.com/ubuntu/ $(lsb_release -cs) main restricted universe multiverse" | sudo tee -a /etc/apt/sources.list
       sudo apt-get update
       sudo apt-get build-dep systemd
-      git clone https://github.com/systemd/systemd-stable --branch v251.2 --depth=1
-      meson systemd-stable/build systemd-stable
-      ninja -C systemd-stable/build systemd-nspawn
-      sudo ln -svf $PWD/systemd-stable/build/systemd-nspawn $(which systemd-nspawn)
+      sudo apt-get install libfdisk-dev
+      git clone https://github.com/systemd/systemd --depth=1
+      meson systemd/build systemd -Drepart=true -Defi=true
+      ninja -C systemd/build
```

Comment:
just build the individual targets needed? repart, nspawn, bootctl, should save some time

Reply:
I was thinking a bit of the future where we'll start using more executables as well, systemd-dissect, systemd-firstboot, maybe tmpfilies ... . It seemed more robust to just build everything

Reply:
Given it runs on every build of every config of every PR, even saving a few minutes will add up, no? Should be easy to see when something new is needed